### PR TITLE
Added mailing with scheduling + resume upload

### PR DESCRIPTION
**Pull request by MohdSharique**
* Added mailers for reminder for interview, update & creation of interview
* Added resume upload
* Added remaining validation for slot clash
Screenshot from 2020-05-26 01-26-54
Screenshot from 2020-05-26 01-28-49
Screenshot from 2020-05-26 01-31-48

### CODE REVIEW

File: app/views/participant_mailer/interview_info.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<p>
+    Hi <%= @participant.email %>, your interview is scheduled on <%= @interview.start_time %>
```
Comment (ayu-111) [inline app/views/participant_mailer/interview_info.html.erb:2]: We can use `strftime` to show time in a specific timezone as +0530

File: app/controllers/participants_controller.rb [hunk 2]
```diff
@@ -6,13 +6,32 @@ def index
...
+      if (interview.start_time <= i.start_time) && (i.start_time <= interview.end_time)
+        isValid = false
+      elsif (interview.start_time <= i.end_time) && (i.end_time <= interview.end_time)
+        isValid = false
+      end
+    end
+
+    if isValid
+      NewInterviewWorker.perform_async(params[:participant][:email], params[:participant][:interview_id])
+      SingleReminderWorker.perform_at(@interview.start_time - 30.minutes, @participant.email)
```
Comment (ayu-111) [inline app/controllers/participants_controller.rb:27]: We should add checks to prevent mail being sent in cases when interview is rescheduled

**Review by ayu-111 — CHANGES_REQUESTED** [review]
Please check comments

File: app/views/participant_mailer/interview_info.html.erb (hunk 1 above)
Comment (MohdSharique) [reply app/views/participant_mailer/interview_info.html.erb]: Sir, please correct me if I'm wrong, I think strftime displays formatted date/time in stored format. After looking up some solutions, I used  .in_time_zone() to convert & display in required local format instead. 
Screenshot from 2020-05-27 23-21-34

File: app/controllers/participants_controller.rb (hunk 2 above)
Comment (MohdSharique) [reply app/controllers/participants_controller.rb:27]: Sir, I have handled this case in reminder_info function of "participant_mailer.rb" file.

File: app/views/participant_mailer/interview_info.html.erb (hunk 1 above)
Comment (ayu-111) [reply app/views/participant_mailer/interview_info.html.erb]: Yes sorry.. I meant to say you can use both of them to modify the timezone and format as well.
This looks good.

File: app/controllers/participants_controller.rb (hunk 2 above)
Comment (ayu-111) [reply app/controllers/participants_controller.rb:27]: But `reminder_info` is not being called from `SingleReminderWorker`. How does that work?

**Pull request merged**